zenodo/modules/utils/cli.py

```diff
@@ -440,3 +445,45 @@ def retry_indexing_failed_openaire_records(eager):
         else:
             openaire_direct_index.delay(
                 str(recid.object_uuid), retry=False)
+
+
+@utils.command('delete_openaire_records')
+@with_appcontext
+def delete_openaire_records():
+    """Deletes half deleted OpenAIRE records."""
+    for key in _iter_openaire_direct_index_keys():
```

Comment:
Since a similar operation is done in the `retry_indexing_failed_openaire_records` command above, I would try to merge them. So the flow is something like:

- Loop over `_iter_openaire_direct_index_keys()`
  - Try to fetch the record
  - If the fetched record/PID is deleted, then run this new workflow added in this PR to delete from OpenAIRE
  - If the record is active, do as before (i.e. try to direct index)